borg/archive.py

```diff
@@ -830,10 +830,10 @@ def report(msg, chunk_id, chunk_no):
                     unpacker.feed(self.key.decrypt(chunk_id, cdata))
                     try:
                         for item in unpacker:
-                            # we expect item to be a dict (and it must also work as StableDict)
-                            if isinstance(item, dict):
+                            # we expect item to be a StableDict and check if .items() works
+                            if isinstance(item, StableDict):
                                 try:
-                                    StableDict(item).items()
+                                    item.items()
```

Comment:
hmm, this raises TypeError if some keys are integer values (due to unpacking going wrong - misaligned, bad unpacker state, ...).

maybe a better check here is to validate all the keys:
```
unknown_keys = set(item) - ITEM_KEYS:
if unknown_keys:
    report & skip
```

BUT: if we do it like that, we must not process a repo made with a more recent borg version that has more valid metadata dict keys.